cmake: extract version string and add `PrintVersion.cmake`

compnerd opened the pull request:
Add a helper to allow us to programatically extract the Swift version string during the build.  This is particularly useful for injection into the packaging stages.

compnerd commented:
@swift-ci please smoke test

compnerd commented on CMakeLists.txt:
```diff
@@ -304,12 +304,7 @@ endif()
-set(SWIFT_VERSION_MAJOR 5)
-set(SWIFT_VERSION_MINOR 9)
-set(SWIFT_VERSION "${SWIFT_VERSION_MAJOR}.${SWIFT_VERSION_MINOR}")
+include(cmake/SwiftVerison.cmake)
```
Yeah ... silly CMake.

compnerd commented:
@swift-ci please smoke test

etcwilde commented on CMakeLists.txt:
```diff
@@ -304,12 +304,7 @@ endif()
-set(SWIFT_VERSION_MAJOR 5)
-set(SWIFT_VERSION_MINOR 9)
-set(SWIFT_VERSION "${SWIFT_VERSION_MAJOR}.${SWIFT_VERSION_MINOR}")
+include(SwiftVerison.cmake)
```
Just `include(SwiftVersion)`

etcwilde replied:
Just `include(SwiftVersion)`

```suggestion
include(SwiftVerison)
```

etcwilde commented on CMakeLists.txt:
```diff
@@ -304,12 +304,7 @@ endif()
-set(SWIFT_VERSION_MAJOR 5)
-set(SWIFT_VERSION_MINOR 9)
-set(SWIFT_VERSION "${SWIFT_VERSION_MAJOR}.${SWIFT_VERSION_MINOR}")
+include(SwiftVerison)
```
```suggestion
include(SwiftVersion)
```

🤦 It would probably help if I could spell.

compnerd commented:
@swift-ci please test

compnerd committed:
cmake: extract version string and add `PrintVersion.cmake`

Add a helper to allow us to programatically extract the Swift version
string during the build. This is particularly useful for injection into
the packaging stages.

Co-authored-by: Evan Wilde <etceterawilde@gmail.com>

compnerd commented:
@swift-ci please test

compnerd commented:
@swift-ci please test

etcwilde reviewed (approved):
LGTM
